dotnet/src/Connectors/Connectors.Amazon/Bedrock/Core/Clients/BedrockChatCompletionClient.cs

```diff
@@ -0,0 +1,182 @@
+        ConverseRequest converseRequest,
+        CancellationToken cancellationToken = default)
+    {
+        // Check that the text from the latest message in the request object is not empty.
+        string? text = converseRequest.Messages?[^1]?.Content?[0]?.Text;
+        if (string.IsNullOrWhiteSpace(text))
+        {
+            throw new ArgumentException("Did not enter proper chat completion message. Text was null or whitespace.");
+        }
+        return await this._bedrockApi.ConverseAsync(converseRequest, cancellationToken).ConfigureAwait(false);
+    }
+    internal async Task<IReadOnlyList<ChatMessageContent>> GenerateChatMessageAsync(
+        ChatHistory chatHistory,
+        PromptExecutionSettings? executionSettings = null,
+        Kernel? kernel = null,
+        CancellationToken cancellationToken = default)
+    {
+        Verify.NotNullOrEmpty(chatHistory);
+        ConverseRequest converseRequest = this._ioService.GetConverseRequest(this._modelId, chatHistory, executionSettings);
+        var regionEndpoint = this._bedrockApi.DetermineServiceOperationEndpoint(converseRequest).URL;
+        this._chatGenerationEndpoint = new Uri(regionEndpoint);
+        ConverseResponse response;
+        using (var activity = ModelDiagnostics.StartCompletionActivity(
+                   this._chatGenerationEndpoint, this._modelId, this._modelProvider, chatHistory, executionSettings))
+        {
+            try
+            {
+                response = await this.ConverseBedrockModelAsync(converseRequest, cancellationToken).ConfigureAwait(false);
+            }
+            catch (Exception ex) when (activity is not null)
+            {
+                activity.SetError(ex);
+                throw;
+            }
+            IEnumerable<ChatMessageContent> chat = ConvertToMessageContent(response);
+            IReadOnlyList<ChatMessageContent> chatMessagesList = chat.ToList();
+            activity?.SetCompletionResponse(chatMessagesList);
+            return chatMessagesList;
+        }
+    }
+    /// <summary>
+    /// Converts the ConverseResponse object as outputted by the Bedrock Runtime API call to a ChatMessageContent for the Semantic Kernel.
+    /// </summary>
+    /// <param name="response"> ConverseResponse object outputted by Bedrock. </param>
+    /// <returns></returns>
+    private static IEnumerable<ChatMessageContent> ConvertToMessageContent(ConverseResponse response)
+    {
+        if (response.Output.Message != null)
+        {
+            var message = response.Output.Message;
+            return new[]
+            {
+                new ChatMessageContent
+                {
+                    Role = MapRole(message.Role.Value),
+                    Items = CreateChatMessageContentItemCollection(message.Content)
+                }
+            };
+        }
+        return Enumerable.Empty<ChatMessageContent>();
+    }
+    private static AuthorRole MapRole(string role)
```

Comment:
We already have a MapRole in `BedrockModelUtilities` (that takes a `AuthorRole` instead of a string). Would it make more sense for this mapper to also live in `BedrockModelUtilities` next to the other MapRole?

Reply:
The `MapRole` in `BedrockModelUtilities` converts `AuthorRole` to `ConversationRole` in order to convert the Semantic Kernel's role attribute to Bedrock's role attribute, thus is a function needed for Bedrock performance. The one in `BedrockChatCompletionClient` is going the other direction (`ConversationRole` converted to `AuthorRole`) to build the Semantic Kernel's desired output, needed for Semantic Kernel performance. I've renamed this MapRole to `MapConversationRoleToAuthorRole` and moved it to a Utilities class for Client objects, and renamed the other MapRole in Bedrock Models file to `MapAuthorRoleToConversationRole`, for clarity (bad naming originally on my part). I think only client objects need `MapConversationRoleToAuthorRole` and only modelIO objects need `MapAuthorRoleToConversationRole`, but if you think they should live in the same file, I can move all utilities to an outermost file and have model and client classes reach out there for needed functions (right now there's a separate client utilities file and a models utilities file).

Reply:
That makes sense to me. The rename helps. Thanks!